### ISSUE TITLE
Newer versions are missing from releases

### ISSUE THREAD

**Issue by IzzySoft** [tzebrowski/ObdGraphs#11]
I see there were recent releases on Google Play (6/2023), while the last release here on Github is one year old. Will you make newer versions available here, too? That would be especially useful to those of us who have no ties to Google at all (de-googled devices) and/or prefer not to use Google Play (e.g. for privacy reasons).

**Comment by IzzySoft** [comment]
@tzebrowski any word?

**Comment by tzebrowski** [comment]
hi @IzzySoft,

Sure, next planned release to Google Play, will be also published on Github.

**Comment by IzzySoft** [comment]
Wonderful, thanks! As the less wonderful news is the date for the delivery of my car was postponed again, I can have some more patience :see_no_evil: Hopefully you'll be attaching the APKs then?

**Comment by IzzySoft** [comment]
Oh, cool: I see the first beta was just published with an APK attached while I wrote the previous comment, thanks! Two questions on it:

    package: name='org.obd.graphs.app' versionCode='1' versionName='1-20220717.1550'

Will you remember to increase `versionCode` with each release, so Android sees newer ones as updates? Otherwise it thinks it already has the latest version.

```
Offending libs:
---------------
* Metrics (/com/codahale/metrics): Tracking

1 offenders.
```

Is tracking on by default? Can it be turned off?

**Comment by tzebrowski** [comment]
Here is the clone of the first production google play release: https://github.com/tzebrowski/ObdGraphs/releases/tag/v74
`wersionCode` is increased each google release, otherwise google rejects artifacts. 
When it comes to `/com/codahale/metrics`, its used for the implementation of `rate collector`  https://github.com/tzebrowski/ObdMetrics/blob/main/src/main/java/org/obd/metrics/diagnostic/DefaultRateCollector.java, cannot be dropped or turn off,

**Comment by IzzySoft** [comment]
> `wersionCode` is increased each google release, otherwise google rejects artifacts.

Right, Google would reject an update with a `versionCode` it already has. I just wonder why the APK attached to the release still has it at `1` – it's not the first release, and [`build.gradle`](https://github.com/tzebrowski/ObdGraphs/blob/master/app/build.gradle) has much higher ones (18 resp. 54 / 74 currently) as well.

As for Codahale, what exactly does it do? According to the library data of my checker it's marked `Tracking`, which implies data is transferred off-device to some 3rd party without the explicit consent of the user ("cannot be dropped or turned off" means there is no expressive consent). Maybe the `Tracking` definition is wrong – but their devs themselves describe the lbrary as "gives you unparalleled insight into what your code does in production", so it's not functional (i.e. not needed for the app to work) but analytical. TL;DR: what you write sounds like it's just used internally (like getting metrics from the car into the app, without transferring it "elsewhere"). Is that correct? Apologies if I might sound a little stupid, but not being an Android dev I'm not "fluent" in reading source code :wink:

**Comment by tzebrowski** [comment]
> Right, Google would reject an update with a versionCode it already has. I just wonder why the APK attached to the release still has it at 1 – it's not the first release, and [build.gradle](https://github.com/tzebrowski/ObdGraphs/blob/master/app/build.gradle) has much higher ones (18 resp. 54 / 74 currently) as well.

Right, that's complex.. Under same project few apps are released to google play..  Anyway, application which is available here https://play.google.com/store/apps/details?id=org.obd.graphs.my.giulia has version 54 assigned, and can be found under Github releases (jar file)  https://github.com/tzebrowski/ObdGraphs/releases/tag/v74

> L;DR: what you write sounds like it's just used internally (like getting metrics from the car into the app, without transferring it "elsewhere"). Is that correct?

Correct.

**Comment by IzzySoft** [comment]
Great, thanks! So to sum up:

* thee APK at releases which currently has `versionCode: 1` will have its `versionCode` increased with each new release
* `/com/codahale/metrics` is only used locally to get the data from the car to the app
* and I should rather stick to the one named `OdbGraphs*release.apk` and ignore the `/^\d+\.apk$/`as otherwise I indeed get a full load of trackers?

`54.apk`:

```
Offending libs:
---------------
* Metrics (/com/codahale/metrics): Tracking
* Crashlytics (/com/crashlytics): NonFreeComp,Tracking
* Google Ads (/com/google/ads): Ads,NonFreeComp
* Firebase Data Transport (/com/google/android/datatransport): NonFreeNet
* Google Mobile Services (/com/google/android/gms): NonFreeComp
* Firebase (/com/google/firebase): NonFreeNet,NonFreeComp
* Firebase Analytics (/com/google/firebase/analytics): NonFreeComp,Tracking
* Firebase Installations (/com/google/firebase/installations): NonFreeNet

8 offenders.
```

That would be a no-go for me and for those using [my repo](https://apt.izzysoft.de/fdroid) :cry: Can we stick to an APK without those (except the one we already clarified that is)?

**Comment by IzzySoft** [comment]
@tzebrowski any chance this can be solved? Santa promised me 4 wheels, and I'm eager to give your app a try then :star_struck:

**Comment by IzzySoft** [comment]
@tzebrowski apologies for nagging again, but as there's no answer yet… Do you plan for a "pure FOSS" version coming without those "8 offenders", or must I rather give up waiting? My 4 wheels are here, the OBD2 Adapter plugged in. But I wouldn't want to use any app with "techbro tracking" :cry:

**Comment by IzzySoft** [comment]
Either all my notifications are going to `/dev/null` – or no answer simply means "no". Is there a chance?

**Comment by tzebrowski** [comment]
I am dealing with different priorities, your request is "somewhere" on my list of TODOs, however not on the top, that's all, no hard feelings, 
If you would like to, you can make a contribution to the project and send PR, I will approve,

**Comment by IzzySoft** [comment]
Thanks! Then I'll have more patience. I'm no Android dev myself, so I unfortunately cannot help out on this with a PR.

**Comment by IzzySoft** [comment]
Seems still to be "further down" on your list, @tzebrowski (and "no hard feelings" either 😉). Checking with the one closed tickets: do I read that correctly that the app is "tied" to Alpha Romeos, and would not work with other cars anyway? In that case, I'd have to give up waiting as it wouldn't work with my car anyway…

**Comment by IzzySoft** [comment]
So is it tied? Do I stand a chance, or should I rather give up?

**Comment by tzebrowski** [comment]
Hi. since the begging, application is dedicated to modern alfa romeo cars.

**Comment by IzzySoft** [comment]
> application is dedicated to modern alfa romeo cars

Thanks Tomek! Then it's not fit for my "cheap car" (I drive a Škoda).

> It could work with other car manufacturers if some makes contributions and provide PIDs list

I wouldn't even know where to start with that 🙈 Quite a pity (for me), as going by the screenshots it looks like the app I never found in the FOSS world yet…

OK, will stop nagging you then – and instead just keep this issue on the watch list for the time we might add it to our repo here (i.e. when those proprietary components might be gone). Wishing you all the best for your wonderful app!
